tests/unit_tests/include/test_helper.hpp

```diff
@@ -53,6 +53,17 @@
             EXPECT_EQ(res, test_passed); \
     } while (0);
 
+// GTEST_SKIP stops the execution of the program.
+// This macro lets a test use multiple EXPECT_TRUE_OR_FUTURE_SKIP and mark a test as skipped only once at the end.
+#define EXPECT_TRUE_OR_FUTURE_SKIP(a, skip) \
+    do {                                    \
+        int res = a;                        \
+        if (res == test_skipped)            \
+            skip = true;                    \
+        else                                \
+            EXPECT_EQ(res, test_passed);    \
+    } while (0);
+
```

Comment:
I think it's not that difficult to change this macro a bit to total things up into passed/failed/skipped tests.
```suggestion
#define EXPECT_TRUE_OR_FUTURE_SKIP(a, skip) \
    do {                                    \
        int res = a;                        \
        if (res == test_skipped)            \
            skip = true;                    \
            skipped_count += 1;             \
        else {                              \
            passed_count += 1;              \
            EXPECT_EQ(res, test_passed);    \
        }                                   \
    } while (0);

```
You might have to return a `pair<int, int>` from your `test_helper` functions though to propagate both the status and the counts. Finally in your `gtest_skip()` call, you can have:
```cpp
    if (skip) {
        // Mark that some tests were skipped
        GTEST_SKIP() << "Passed: " << passed_count << ", Skipped: " << skipped_count << std::endl;
    }
```

Reply:
I have done that in https://github.com/oneapi-src/oneMKL/pull/407/commits/7954345dcda900b24c1dcf4678b060ac3dd63303